flink-autoscaler/src/main/java/org/apache/flink/autoscaler/tuning/MemoryTuning.java

```diff
@@ -277,6 +286,32 @@ private static MemorySize adjustNetworkMemory(
         return new MemorySize(memBudget.budget(maxNetworkMemory));
     }
 
+    /**
+     * Calculate how many network segment current vertex needs.
+     *
+     * @param currentVertexParallelism The parallelism of current vertex.
+     * @param otherVertexParallelism The parallelism of other vertex.
+     */
+    @VisibleForTesting
+    static int calculateNetworkSegmentNumber(
+            int currentVertexParallelism,
+            int otherVertexParallelism,
+            String shipStrategy,
+            int buffersPerChannel,
+            int floatingBuffers) {
+        // TODO When the parallelism is changed via the rescale api, the FORWARD may be changed to
+        // RESCALE. This logic may needs to be updated after FLINK-33123.
+        if (currentVertexParallelism == otherVertexParallelism && "FORWARD".equals(shipStrategy)) {
+            return buffersPerChannel + floatingBuffers;
+        } else if ("FORWARD".equals(shipStrategy) || "RESCALE".equals(shipStrategy)) {
+            final int channelCount =
+                    (int) Math.ceil(1.0d * otherVertexParallelism / currentVertexParallelism);
```

Comment:
I prefer the following to avoid adding another constant. 
```suggestion
                    (int) Math.ceil(otherVertexParallelism / (double) currentVertexParallelism);
```